ananse/influence.py

```diff
@@ -259,40 +394,96 @@ def plot_influscore(infile, outfile):
+        if union_grns == True:
+            if (GRN_source_file is None) or (GRN_target_file is None):
+                logger.warning(
+                    "You should provide at least two ANANSE network files to take the interaction union!"
+                )
+            else:
+                logger.info(
+                    f"Reading network(s), using the union of the top {edges} edges of each GRN."
+                )
+                top_int_source = read_top_interactions(GRN_source_file, edges=edges)
+                top_int_target = read_top_interactions(GRN_target_file, edges=edges)
+                top_int = set.union(top_int_source, top_int_target)
+                G1_source = read_network(GRN_source_file, interactions=top_int)
+                G2_target = read_network(GRN_target_file, interactions=top_int)
+        self.G = difference(G1_source, G2_target)
+        logger.info(f"Differential network has {len(self.G.edges)} edges.")
         # Load expression file
         self.expression_change = read_expression(degenes)
-
         self.outfile = outfile
-
         # Filter TFs
         self.filter = filter
 
     def save_reg_network(self, filename):
         """Save the network difference between two cell types to a file."""
-
+        full_output_GRN = bool(nx.get_edge_attributes(self.G, "tf_expr_source"))
         with open(filename, "w") as nw:
-            for (u, v, d) in self.G.edges(data=True):
-                nw.write(u + "\t" + v + "\t" + str(d["weight"]) + "\n")
+            if full_output_GRN:
+                logger.info("output full diff network")
+                nw.write(
+                    f"TF \t target \t weight \t source_weight \t target_weight \t tf_expr_source \t tf_expr_target \t tg_expr_source \t tg_expr_target \t source_wb \t target_wb \t source_Tf_act \t target_Tf_act \n"
+                )
+                for (u, v, d) in self.G.edges(data=True):
+                    nw.write(
+                        u
+                        + "\t"
+                        + v
+                        + "\t"
+                        + str(d["weight"])
+                        + "\t"
+                        + str(d["source_weight"])
+                        + "\t"
+                        + str(d["target_weight"])
+                        + "\t"
+                        + str(d["tf_expr_source"])
+                        + "\t"
+                        + str(d["tf_expr_target"])
+                        + "\t"
+                        + str(d["tg_expr_source"])
+                        + "\t"
+                        + str(d["tg_expr_target"])
+                        + "\t"
+                        + str(d["source_wb"])
+                        + "\t"
+                        + str(d["target_wb"])
+                        + "\t"
+                        + str(d["TF_act_source"])
+                        + "\t"
+                        + str(d["TF_act_target"])
+                        + "\n"
+                    )
```

Comment:
```
cols = [u, v, d["weight"], d["source_weight", ... ]]
"\t".join(cols) + "\n"
```

this will auto-convert everything to string, and its more readable